package.json

```diff
@@ -100,22 +100,23 @@
     "compression": "^1.7.0",
     "connect-history-api-fallback": "^1.3.0",
     "express": "^4.15.3",
-    "long": "^3.2.0",
+    "long": "^4.0.0",
     "minimist": "^1.2.0",
-    "mobx": "^3.3.1",
-    "mobx-react": "^4.3.3",
+    "mobx": "^4.1.0",
+    "mobx-react": "^5.0.0",
+    "mobx-utils": "^4.0.0",
     "nuclearnet.js": "^1.5.2",
     "protobufjs": "6.7.3",
     "react": "^16.2.0",
     "react-color": "^2.13.8",
     "react-dom": "^16.0.0",
-    "react-resize-detector": "^1.1.0",
+    "react-resize-detector": "^2.2.0",
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
     "serve-favicon": "^2.4.3",
-    "socket.io": "^1.7.3",
-    "socket.io-client": "^1.7.3",
-    "three": "^0.84.0",
+    "socket.io": "^2.0.4",
+    "socket.io-client": "^2.0.4",
```

Comment:
The current types are only for `1.4.29`.

Reply:
Not a problem, the 2.0 was a change to the wire protocol etc rather than a change to the API (mainly) and the performance improvements are worth it.

Reply:
Also the types for 2.0 don't exist yet

Reply:
There are definitely API changes that the old types won't let you use.

> performance improvements are worth it.

Have you tested that? I'm not sure simply upgrading to 2.0 has any wire protocol changes? IIRC you needed to explicitly import/use some new thing that the types don't know exists.

Reply:
I don’t think so, the changelog implied it changed the defaults. And merged the negotiation packets on the wire. The only breaking changes are irrelevant to us.

Reply:
Mmmk. 🤷‍♂️ I hope this doesn't cause someone a lot of confusion one day!

Reply:
Well Socket.IO version 2.0 has been out for almost a year at this point... If the types haven't been updated in that time it implies that either nobody is using socket.io or there isn't anything to change. (or people are lazy)